api.swagger.yaml

```diff
@@ -467,96 +576,803 @@ paths:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The user has been removed from the group.
+        '400':
+          description: The UUID was malformed.
+        '404':
+          description: |
+            The user or group does not exist, or the user was not a member
+            of the group.
+      description: Remove the user's membership of the group.
+    put:
+      tags:
+        - groups
+      parameters:
+        - description: The Invotra UUID of the group.
+          in: path
+          name: group_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+        - description: The Invotra UUID of the user.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The user's membership of this group was updated.
+        '201':
+          description: The user was added as a member of the group.
+        '400':
+          description: The UUID was malformed or the role type was invalid.
+        '404':
+          description: The user or group does not exist.
+      description: >-
+        Add a user to a group or update information about a user's existing
+        group membership.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/group-membership-common-schema'
+  /posts:
+    get:
+      tags:
+        - posts
+      parameters:
+        - description: Filter by external ID.
+          in: query
+          name: external_id
+          schema:
+            type: string
+      responses:
+        '200':
+          description: >
+            List of posts matching the filter criteria, or an empty array if
+            none were found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/uuid-list-schema'
+      description: Search for posts.
+    post:
+      tags:
+        - posts
+      responses:
+        '201':
+          description: The post was created successfully.
+          headers:
+            Location:
+              description: |
+                The URL at which the newly provisioned post may be retrieved.
+              schema:
+                type: string
+        '400':
+          description: The post was not saved due to malformed JSON.
+        '422':
+          description: |
+            The post was not saved due to Drupal validation failure or JSON
+            schema validation failure.
+      description: Provision a new post.
```

Comment:
I think "provision" is a strange word to use in this context :)

Reply:
updated